### ISSUE TITLE
508-defect-2: Military service information field uses the maxlength attribute without a hint or validation

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#48234]
### Point of contact

Josh Kim

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

The military service information field uses the maxlength attribute to cap input to 9 characters. While this prevent users from inputting beyond 9 characters, no feedback, hint, or validation is actually provided to them to know that. 

For family members and caregives who may not be familiar with this, this may be even more difficult for them to understand why they can't continue typing into the field.

<img width="1491" alt="Screen Shot 2022-10-14 at 10 06 26 AM" src="https://user-images.githubusercontent.com/14154792/195867414-e5255d87-d7e0-4135-a772-fa34395986ea.png">

### Reproduction steps

1. [Start a pre-need application without signing in](https://www.va.gov/burials-and-memorials/pre-need/form-10007-apply-for-eligibility/sponsor-information)
2. Select "other" under "relationship to service member"
3. Attempt to type 10 characters into the military service number field and confirm that it caps at 9
4. Then, type in "123" into the field, hit continue and confirm no validation is provided
...

### Proposed solution or next steps

Ideally, we'd provide two things to the user:

- [ ] Static label or hint text informing the user of the expected character length 
- [ ] On submit validation (upon clicking continue) the user should be notified why their input failed and how to fix it (e.g. "A military service number is usually X characters long")

<img width="536" alt="Screen Shot 2022-10-14 at 10 19 30 AM" src="https://user-images.githubusercontent.com/14154792/195869663-d1f2dca8-b105-417d-9b96-88ced57d1812.png">

@jsokohl to review 🙏 

### References, articles, or WCAG support

1. [Live validation is problematic](https://adamsilver.io/blog/live-validation-is-problematic/)
2. [Form design: zero to hero](https://adamsilver.io/blog/form-design-from-zero-to-hero-all-in-one-blog-post/#crafting-questions)

### Type of issue

- [X] Screenreader
- [X] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by jaredcunha** [comment]
Seeing a pattern here and perhaps there might be a need get a design system component to use hint text similar to US web design system… unless that's already built, but I can't find any documented examples of it.

**Comment by joshkimux** [comment]
It's a little hidden, but [it's under the text input component](https://design.va.gov/components/form/#hint-text)

**Comment by jaredcunha** [comment]
Grrr - that uses utility classes and does not make use of `aria-describedby`. Regardless, that pattern might be in the form system. But I could be wrong. But I could be right?

**Comment by jsokohl** [comment]
Why not change the label to "Spouse's nine-digit military service number...."?

**Comment by jaredcunha** [comment]
@jsokohl That often ends up being the type pattern I've seen in lieu of USWDS-style hint text. But I do think it might be worth getting this into the component given how it could provide more helpful designs across VA teams.

**Comment by ccruzgra** [comment]
@joshkimux It looks like this is also lacking validation on the Service Member/Veteran flow, which has the same field without the sponsor language.
See:
<img width="556" alt="Screenshot 2023-02-27 at 1 49 36 PM" src="https://user-images.githubusercontent.com/41477797/221655644-92c87fb2-ca23-40c8-953a-86a457d1f446.png">

**Comment by ccruzgra** [comment]
@coforma-terry We have an idea of the requirements that this validation message should capture. What is the best way to go about requesting content be drafted for this?

**Comment by coforma-terry** [comment]
Hi @ccruzgra - I think it's fine for you to draft this yourselves if you've reviewed the content style guide standards.

**Comment by ccruzgra** [comment]
@coforma-terry Sounds great. Do small content changes like this that we draft still need to go through some approval process or are we good to implement?

**Comment by coforma-terry** [comment]
Hey @ccruzgra - No, you can implement that yourselves :)

**Comment by sonntag-adam** [comment]
@coforma-terry @SarahKay8 fix is in staging.

**Comment by ccruzgra** [comment]
@SarahKay8 based on our requirements, this needs to indicate "characters", not "digits" since acceptable entries could be alphanumeric. The validation messaging should display for 3 digits or less since the acceptable range is 4-10 characters.

**Comment by SarahKay8** [comment]
Thanks! @ccruzgra Closing now 🌱
